From 45c6749a31fb2e9775423618bee9d5f5b572bc42 Mon Sep 17 00:00:00 2001
From: Thomas Gleixner <tglx@linutronix.de>
Date: Fri, 4 Nov 2011 18:58:24 +0100
Subject: [PATCH 230/271] hotplug-stuff.patch

Do not take lock for non handled cases (might be atomic context)

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 kernel/workqueue.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/kernel/workqueue.c b/kernel/workqueue.c
index 8389afe..674d783 100644
--- a/kernel/workqueue.c
+++ b/kernel/workqueue.c
@@ -3219,6 +3219,11 @@ static int __devinit workqueue_cpu_up_callback(struct notifier_block *nfb,
 		new_worker = create_worker(gcwq, false);
 		if (!new_worker)
 			return NOTIFY_BAD;
+	case CPU_UP_CANCELED:
+	case CPU_ONLINE:
+		break;
+	default:
+		return notifier_from_errno(0);
 	}
 
 	/* some are called w/ irq disabled, don't disturb irq status */
-- 
1.7.10

